root/static/scripts/edit/components/Dialog.js

```diff
@@ -0,0 +1,182 @@
+  const preventOverflow =
+    require('@popperjs/core/lib/modifiers/preventOverflow').default;
+
+  return {
+    modifiers: [
+      {
+        enabled: false,
+        name: 'eventListeners',
+      },
+      flip,
+      preventOverflow,
+      ...arrowModifiers,
+    ],
+    placement: 'right',
+  };
+};
+
+function getDialogNode(dialogRef, dialogId) {
+  let dialogNode = dialogRef.current;
+  if (typeof document !== 'undefined' && !dialogNode) {
+    dialogNode = document.getElementById(dialogId);
+    if (!dialogNode) {
+      dialogNode = document.createElement('div');
+      dialogNode.setAttribute('id', dialogId);
+      dialogNode.classList.add('popper-dialog');
+      document.body?.appendChild(dialogNode);
+    }
+    dialogRef.current = dialogNode;
+  }
+  return dialogNode;
+}
+
+const Dialog = ({
+  accept,
+  cancel,
+  children,
+  closeOnOutsideClick = true,
+  dialogId,
+  id,
+  hasErrors,
+  opener,
+  showArrow = true,
+}: Props): React.Node => {
+  const containerRef = React.useRef(null);
+  const dialogRef = React.useRef(null);
+
+  React.useLayoutEffect(() => {
+    const {createPopper} = require('@popperjs/core/lib/popper-lite');
+
+    const dialogNode = getDialogNode(dialogRef, dialogId);
+    if (!dialogNode) {
+      throw new Error('dialogRef is null');
+    }
+
+    dialogNode.style.visibility = 'visible';
+
+    const popper = createPopper(
+      opener.current,
+      dialogNode,
+      getPopperOptions(showArrow),
+    );
+
+    const container = containerRef.current;
+    if (container) {
+      const toFocus = container.querySelector('input, select');
+      (toFocus ?? container).focus({
+        preventScroll: true,
+      });
+    }
+
+    return () => {
+      dialogNode.style.visibility = 'hidden';
+      popper.destroy();
+    };
+  }, [dialogId, opener, showArrow]);
+
+  const handleInputKeyDown = React.useCallback((
+    event: SyntheticKeyboardEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    if (event.defaultPrevented) {
+      return false;
+    }
+
+    const nodeName = event.target instanceof HTMLElement
+      ? event.target.nodeName.toLowerCase()
+      : '';
+    if (event.keyCode === 13 && /^input|select$/.test(nodeName)) {
+      if (!hasErrors) {
+        event.stopPropagation();
+        accept(true /* returnFocusToOpener */);
+      }
+    } else if (event.keyCode === 27) {
+      event.stopPropagation();
+      cancel(true /* returnFocusToOpener */);
+    }
+
+    return true;
+  }, [accept, cancel, hasErrors]);
+
+  const dialogNode = getDialogNode(dialogRef, dialogId);
+
+  useOutsideClickEffect(
+    dialogRef,
+    closeOnOutsideClick ? ((target) => {
+      if (target !== opener.current) {
+        cancel(false);
+      }
+    }) : null,
+  );
+
+  return dialogNode ? createPortal(
+    <>
+      <div
+        id={id}
+        onKeyDown={handleInputKeyDown}
+        ref={containerRef}
+        tabIndex="-1"
+      >
```

Comment:
FWIW, at least with the example you give here, I feel this div (so, the whole dialog content including the button) could use a bit of left padding not to be so close to the left edge of the popup :) 5 to 10 px probably?